### PR TITLE
Refactor Unit repo, service and e2e tests

### LINKED ISSUE #334: Refactor unit repo, service and e2e tests
Look through the new refactored tests (see Systems, Manufacturers and Catalogue Categories) and refactor the unit and e2e tests for units to match.

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

see #334

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
closes #334

### CODE REVIEW

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Looks good, just some minor things and recent changes I made in my latest PRs.

File: test/e2e/test_unit.py [hunk 1]
```diff
@@ -1,179 +1,238 @@
...
 
 
-def test_delete(test_client):
-    """Test deleting a unit"""
+class DeleteDSL(ListDSL):
+    """Base class for delete tests."""
 
-    response = test_client.post("/v1/units", json=UNIT_POST_A)
-    unit = response.json()
+    _delete_response: Response
```
Comment (joelvdavies) [inline test/e2e/test_unit.py:174]:
```suggestion
    _delete_response_unit: Response
```

**Review by joelvdavies — CHANGES_REQUESTED** [review]

File: test/e2e/test_unit.py [hunk 2]
```diff
@@ -1,179 +1,238 @@
...
+    def test_create_unit(self):
+        """Test creating a unit"""
 
+        self.post_unit(UNIT_POST_DATA_MM)
+        self.check_post_unit_success(UNIT_GET_DATA_MM)
 
-def test_create_unit_with_duplicate_name(test_client):
-    """Test creating a unit with a duplicate name"""
+    def test_create_unit_with_duplicate_name(self):
+        """Test creating a unit with a duplicate name"""
```
Comment (joelvdavies) [inline test/e2e/test_unit.py:76]:
```suggestion
    def test_create_unit_with_duplicate_value(self):
        """Test creating a unit with a duplicate value."""
```

**Comment by joelvdavies** [comment]
For ease of merging later, could you rebase this branch on top of items in https://github.com/ral-facilities/inventory-management-system-api/pull/359 please.

File: test/unit/services/test_unit.py [hunk 3]
```diff
@@ -2,102 +2,187 @@
...
+        self._expected_unit_in = UnitIn(**unit_post_data, code=utils.generate_code(unit_post_data["value"], "unit"))
+        self._expected_unit_out = UnitOut(**self._expected_unit_in.model_dump(), id=ObjectId())
+
+        ServiceTestHelpers.mock_create(self.mock_unit_repository, self._expected_unit_out)
+
+    def call_create(self) -> None:
+        """
+        Calls the `UnitService` `create` method with the appropriate data from a prior call to
+        `mock_create`.
+        """
```
Comment (VKTB) [inline test/unit/services/test_unit.py:69]:
```suggestion
        """Calls the `UnitService` `create` method with the appropriate data from a prior call to `mock_create`."""
```

**Review by VKTB — COMMENTED** [review]
Just some very minor things, it looks good otherwise.

**Review by joelvdavies — APPROVED** [review]

**Review by VKTB — APPROVED** [review]

**Pull request merged**